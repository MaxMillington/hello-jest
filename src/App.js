import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import MyComp from './MyComp'

class App extends Component {
    state = { click: false }

  myClickFunc = () => {
      this.setState({ click: true })
      console.log('clickity clickcty')
  }

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
          <MyComp click={this.myClickFunc} />
      </div>
    );
  }
}

export default App;
