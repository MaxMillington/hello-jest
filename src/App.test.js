import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount, render } from 'enzyme'

describe('my sweet test', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        expect(div).toMatchSnapshot()
    });
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        expect(div).toMatchSnapshot()
    });
})

describe('my sweet test', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        expect(div).toMatchSnapshot()
    });
    it('renders without crashing', () => {
        const app = mount(<App />)
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        expect(div).toMatchSnapshot()
        expect(app.find('.App-intro').text()).toContain('To get started, edit src/App.js and save to reload.')
    });
    it('clicks it hard corr', () => {
        const app = mount(<App />)
        const instance = app.instance()
        const spy = jest.spyOn(instance, 'myClickFunc')

        const p = app.find('.App-intro')
        p.simulate('click')
        expect(app.state()).toEqual({click: true})
    })
})
