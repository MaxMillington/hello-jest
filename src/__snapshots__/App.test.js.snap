// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my sweet test renders without crashing 1`] = `
<div>
  <div
    data-reactroot=""
    class="App"
  >
    <div
      class="App-header"
    >
      <img
        src="logo.svg"
        class="App-logo"
        alt="logo"
      />
      <h2>
        Welcome to React
      </h2>
    </div>
    <div>
      <p
        class="App-intro"
      >
        <!-- react-text: 7 -->
        To get started, edit 
        <!-- /react-text -->
        <code>
          src/App.js
        </code>
        <!-- react-text: 9 -->
        and save to reload.
        <!-- /react-text -->
      </p>
    </div>
  </div>
</div>
`;

exports[`my sweet test renders without crashing 2`] = `
<div>
  <div
    data-reactroot=""
    class="App"
  >
    <div
      class="App-header"
    >
      <img
        src="logo.svg"
        class="App-logo"
        alt="logo"
      />
      <h2>
        Welcome to React
      </h2>
    </div>
    <div>
      <p
        class="App-intro"
      >
        <!-- react-text: 7 -->
        To get started, edit 
        <!-- /react-text -->
        <code>
          src/App.js
        </code>
        <!-- react-text: 9 -->
        and save to reload.
        <!-- /react-text -->
      </p>
    </div>
  </div>
</div>
`;

exports[`my sweet test renders without crashing 3`] = `
<div>
  <div
    data-reactroot=""
    class="App"
  >
    <div
      class="App-header"
    >
      <img
        src="logo.svg"
        class="App-logo"
        alt="logo"
      />
      <h2>
        Welcome to React
      </h2>
    </div>
    <div>
      <p
        class="App-intro"
      >
        <!-- react-text: 7 -->
        To get started, edit 
        <!-- /react-text -->
        <code>
          src/App.js
        </code>
        <!-- react-text: 9 -->
        and save to reload.
        <!-- /react-text -->
      </p>
    </div>
  </div>
</div>
`;

exports[`my sweet test renders without crashing 4`] = `
<div>
  <div
    data-reactroot=""
    class="App"
  >
    <div
      class="App-header"
    >
      <img
        src="logo.svg"
        class="App-logo"
        alt="logo"
      />
      <h2>
        Welcome to React
      </h2>
    </div>
    <div>
      <p
        class="App-intro"
      >
        <!-- react-text: 7 -->
        To get started, edit 
        <!-- /react-text -->
        <code>
          src/App.js
        </code>
        <!-- react-text: 9 -->
        and save to reload.
        <!-- /react-text -->
      </p>
    </div>
  </div>
</div>
`;
